js/src/main/scala/crystal/react/hooks/UseAsyncEffectOnMount.scala

```diff
@@ -0,0 +1,77 @@
+package crystal.react.hooks
+
+import cats.syntax.all._
+import crystal.react.implicits._
+import japgolly.scalajs.react._
+import japgolly.scalajs.react.hooks.CustomHook
+import japgolly.scalajs.react.util.DefaultEffects.{ Async => DefaultA }
+
+// TODO PR to scalajs-react changing implicit not found message
+object UseAsyncEffectOnMount {
+
+  val hook = CustomHook[DefaultA[DefaultA[Unit]]]
+    .useRef(none[DefaultA[Unit]])
+    .useEffectOnMountBy((effect, cleanup) =>
+      effect
+        .flatMap(f => cleanup.setAsync(f.some))
+        .runAsyncAndForget
+        .as(cleanup.get.flatMap(_.foldMap(_.runAsyncAndForget)))
+    )
+    .build
+
+  object HooksApiExt {
+    sealed class Primary[Ctx, Step <: HooksApi.AbstractStep](api: HooksApi.Primary[Ctx, Step]) {
+
+      /** Simulates `useEffect` with cleanup callback for async effect. To declare an async effect
```

Comment:
Comments in this file are the same as in `UseAsyncEffect`.

Reply:
Thank you, fixed.